docs/modules/password_auth_provider_callbacks.md

```diff
@@ -0,0 +1,155 @@
+# Password auth provider callbacks
+
+Password auth providers offer a way for server administrators to integrate
+their Synapse installation with an external authentication system. The callbacks can be
+registered by using the Module API's `register_password_auth_provider_callbacks` method.
+
+## Callbacks
+
+### `auth_checkers`
+
+```
+ auth_checkers: Dict[Tuple[str,Tuple], CHECK_AUTH_CALLBACK]
```

Comment:
I think this line is confusing to people not familiar with this portage work.

I would suggest:

* replacing `CHECK_AUTH_CALLBACK` with `Callable` (which is the generic Python type for a function)
* moving the example at line 41 (and the few paragraphs below it) up to ~line 18 (and having that paragraph end with something like "An authentication checking callback should be of the following form:")